ui/raidboss/data/07-dt/trial/zelenia-ex.ts

```diff
@@ -1,14 +1,638 @@
+
+  const entry = Object.entries(mapEffectData).find((entry) => entry[1].location === loc);
+
+  if (entry === undefined)
+    throw new UnreachableCode();
+
+  const [key] = entry;
+
+  if (key === undefined)
+    throw new UnreachableCode();
+
+  return key as MapEffectTile;
+};
+
+const headMarkerData = {
+  // Thorns tether purple markers during Bloom 4
+  'thorns': '000C',
+  // Adds red headmarker showing you're tethered
+  'addsTether': '0017',
+  // Escelon 2 stack marker
+  'fourPlayerStack': '005D',
+  // Bloom 1 clockwise rotation indicator
+  'clockwise': '00A7',
+  // Bloom 1 counterclockwise rotation indicator
+  'counterclockwise': '00A8',
+  // "Shock" donut marker
+  'shockDonut': '0244',
+  // "Shock" spread marker
+  'shockSpread': '0245',
+  // "Stock Break" multi-hit marker
+  'fiveHitStack': '024E',
+  // Rose/Flower headmarker
+  'roseFlower': '0250',
+  // Rose spread marker
+  'roseSpread': '0254',
+  // Stack marker during Bloom 4, for "Alexandrian Banish III"
+  'bloom4Stack': '0255',
+} as const;
+
+const defaultTileState = () => ({
+  bloomTileInnerNNE: 'unknown',
+  bloomTileInnerENE: 'unknown',
+  bloomTileInnerESE: 'unknown',
+  bloomTileInnerSSE: 'unknown',
+  bloomTileInnerSSW: 'unknown',
+  bloomTileInnerWSW: 'unknown',
+  bloomTileInnerWNW: 'unknown',
+  bloomTileInnerNNW: 'unknown',
+  bloomTileOuterNNE: 'unknown',
+  bloomTileOuterENE: 'unknown',
+  bloomTileOuterESE: 'unknown',
+  bloomTileOuterSSE: 'unknown',
+  bloomTileOuterSSW: 'unknown',
+  bloomTileOuterWSW: 'unknown',
+  bloomTileOuterWNW: 'unknown',
+  bloomTileOuterNNW: 'unknown',
+} as const);
+
+export interface Data extends RaidbossData {
+  escelonFallBaits: ('near' | 'far')[];
+  phase: Phase;
+  tileState: {
+    [loc in MapEffectTile]: 'unknown' | 'red' | 'grey';
+  };
+  bloom1StartDir?: number;
+}
 
 const triggerSet: TriggerSet<Data> = {
   id: 'RecollectionExtreme',
   zoneId: ZoneId.RecollectionExtreme,
   timelineFile: 'zelenia-ex.txt',
-  triggers: [],
+  initData: () => ({
+    escelonFallBaits: [],
+    phase: 'phase1',
+    tileState: { ...defaultTileState() },
+  }),
+  triggers: [
+    {
+      id: 'ZeleniaEx Tile Tracker',
+      type: 'MapEffect',
+      netRegex: {
+        location: tileSlots,
+        flags: [bloomTileFlags.red, bloomTileFlags.grey, bloomTileFlags.greyToRed],
+        capture: true,
+      },
+      run: (data, matches) => {
+        let newState: 'unknown' | 'red' | 'grey' = 'unknown';
+
+        switch (matches.flags) {
+          case bloomTileFlags.red:
+          case bloomTileFlags.greyToRed:
+            newState = 'red';
+            break;
+          case bloomTileFlags.grey:
+            newState = 'grey';
+            break;
+        }
+        data.tileState[getTileNameFromLocation(matches.location)] = newState;
+      },
+    },
+    {
+      id: 'ZeleniaEx Phase Tracker',
+      type: 'StartsUsing',
+      netRegex: {
+        id: [
+          'A8AD',
+          'A8B5',
```

Comment:
Rather than do this with a list of ids in the netregex and a `switch` block in the trigger, I'd suggest creating a map like is done here:
https://github.com/OverlayPlugin/cactbot/blob/9a285de71f351d99baab27a3f0ae98be2468f0ff/ui/raidboss/data/07-dt/ultimate/futures_rewritten.ts#L19-L29

and then doing a more simplified trigger for phase tracking, like:

https://github.com/OverlayPlugin/cactbot/blob/9a285de71f351d99baab27a3f0ae98be2468f0ff/ui/raidboss/data/07-dt/ultimate/futures_rewritten.ts#L474-L479

Reply:
That doesn't really work because of the required conditional for `escalon1` and `escalon2`.

`escalon1` could be safely dropped as a phase and just bundled into `phase1`, but I think we might need `escalon2` and `escalon3` to be split out. If that turns out to not be the case later on down the line, then yes, this should be simplified.